File: app/src/main/java/com/aceplus/hackthon/login/LoginPresenter.java
package com.aceplus.hackthon.login;

import com.aceplus.shared.VO.UserVO;
import com.aceplus.shared.model.BackendModel;
import com.aceplus.shared.modelcallback.ModelCallback;

import org.jetbrains.annotations.NotNull;

/**
 * Created by kyawthetwin on 9/22/18.
 */

public class LoginPresenter implements LoginContract.Presenter {

    LoginContract.View view;

    public LoginPresenter(LoginContract.View view) {
        this.view = view;
    }

    @Override
    public void addUser(UserVO userVO) {
        view.showLoading(true);
        BackendModel.Companion.getInstance().addUser(userVO, new ModelCallback.LoginUserCallback() {
            @Override
            public void loginFailed(@NotNull String message) {
                view.showLoading(false);
            }

            @Override
            public void loginSucceed(@NotNull UserVO userVO) {
                view.showLoading(false);
            }
        });
    }
}

File: backend/src/main/java/mvp/presenter/LoginPresenter.kt
package mvp.presenter

import android.text.TextUtils
import com.aceplus.shared.model.BackendModel
import com.aceplus.shared.modelcallback.ModelCallback
import mvp.view.LoginView

/**
 * Created by kkk on 9/22/2018.
 */
class LoginPresenter(loginView: LoginView) : BasePresenter<LoginView>(loginView) {
    fun loginAdmin(username: String, password: String) {
        if (TextUtils.isEmpty(username) || TextUtils.isEmpty(password)) {
            mView.displayMessage("Check UserName Or Password")
        } else {
            BackendModel.getInstance().loginAdmin(username, password, object : ModelCallback.LoginAdminCallback {
                override fun loginFailed(message: String) {
                    mView.displayMessage("Cannot Login")
                }

                override fun loginSucceed(userId: String) {
                    mView.navigateToMainActivity()
                }

            })
        }
    }
}